include/revng/Pipes/TemporaryFile.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+#include <optional>
+
+#include "revng/Pipeline/ContainerSet.h"
+#include "revng/Pipeline/LLVMContainerFactory.h"
+#include "revng/Pipeline/Target.h"
+#include "revng/Support/Assert.h"
+
+namespace pipeline {
+
+/// A temporary file is a container that wraps a single temporary file on disk.
+/// The enumeration of this container will yield the empty set if there is
+/// currently no file associated to a instance of Temporary file, and will
+/// return The target ("root", K) otherwise, where K is the kind provided at
+/// construction time.
+class TemporaryFile : public Container<TemporaryFile> {
```

Comment:
Is it correct to say that the main use case for this is materializing files for command line tools we launch (e.g., linker) that are not in memory yet?

Once all of these are in-memory, we wouldn't have much use for this, is this correct?

Or is this to arbitrarily serialize generic files?

Reply:
more like, some stuff we do uses executables that we cannot use as a library, (linker, merge dynamic) and they need a file do to stuff on them. furthermore it's used by the input and output too.

Reply:
Mmh, then it's more `File` than `TemporaryFile`. Maybe `FileContainer`? The input and the output are not *temporary*.

Reply:
they are temporary in the sense that while executing are temporary, the input, output and those written into steps are the serialized non temporary version,

Reply:
Mmmh, intuitively if I serialize a *temporary file* I still get a temporary file.
Just as if I serialize an LLVM module I still get an LLVM module.
I feel like it gives the wrong intuition, but I also understand the parallel with the fact that `llvm::Module` in memory is "temporary" too.